src/components/Api.js

```diff
@@ -0,0 +1,74 @@
+export default class Api {
+  constructor(baseURL, headers) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    return res.ok ? res.json() : Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseURL}/cards`, {
+      headers: { ...this._headers },
+    }).then(this._checkResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "GET",
+      headers: { ...this._headers },
+    }).then(this._checkResponse);
+  }
+
+  setUserInfo(name, about) {
+    return fetch(`${this._baseURL}/users/me`, {
+      method: "PATCH",
+      headers: {
+        ...this._headers,
+        "Content-Type": "application/json",
```

Comment:
`        "Content-Type": "application/json",
`

`headers: this._headers,` should be used everywhere not to duplicate the same headers